File: c/src/Behavioral/Strategy/DvdNameReplaceSpacesStrategy.c

//DvdNameReplaceSpacesStrategy - three of three concrete strategies

#include "DvdNameStrategy.h"

#include "../../common/common_str.h"


char * DvdNameReplaceSpacesStrategy_formatDvdName(char * src,char charIn) 
{
	return common_str_replace( src, ' ',charIn);
}


DvdNameStrategy_t * DvdNameReplaceSpacesStrategy_new()
{
	DvdNameStrategy_t * s = DvdNameStrategy_new();
	s->formatDvdName = DvdNameReplaceSpacesStrategy_formatDvdName;
	return s;
}

File: php/src/Behavioral/Strategy/DvdNameReplaceSpacesStrategy.php
<?

#//DvdNameReplaceSpacesStrategy - three of three concrete strategies

require_once 'DvdNameStrategy.php';


class DvdNameReplaceSpacesStrategy implements DvdNameStrategy
{
	function formatDvdName($name,$char)
	{
		return str_replace(' ',$char,$name);
	}
}



